bilby/gw/likelihood.py

```diff
@@ -1452,17 +1454,26 @@ def __init__(self, interferometers, waveform_generator,
         self.per_detector_fiducial_waveforms = {}
         self.bin_freqs = dict()
         self.bin_inds = dict()
-        self.initial_parameter_keys_sorted = sorted(self.initial_parameters)
-        self.maximum_likelihood_parameters = initial_parameters
-        self.set_fiducial_waveforms(self.initial_parameters)
+        self.set_fiducial_waveforms(self.fiducial_parameters)
         logger.info("Initial fiducial waveforms set up")
         self.setup_bins()
         self.compute_summary_data()
         logger.info("Summary Data Obtained")
 
+        if update_fiducial_parameters:
+            # write a check to make sure prior is not None
```

Comment:
It would be good to have a `logger` statement here:

`logger.info("Using scipy maximize likelihood and update fiducial parameters.")`